### ISSUE TITLE
bug: "Timestamp cmp Varchar" error when comparing timestamp field with "to_timstamp"

### ISSUE THREAD

**Issue by hzxa21** [risingwavelabs/risingwave#7736]
### Describe the bug

When testing risingwave (commit: `b5c154d8a62ecf646207f09f2d2792cb5c678160`) integration with superset (commit `79114bcd29b0ce1685828ff9a60791d13a7459c9`), it failed to create time series chart and threw " QueryError: internal error: Failed to build executor: Expr error: Unsupported function: Timestamp cmp Varchar" error for the following query:
```sql
SELECT v1 AS __timestamp,
v1 AS v1,
COUNT(*) AS count
FROM public.mv1
WHERE v1 >= TO_TIMESTAMP('2023-02-06 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US')
AND v1 < TO_TIMESTAMP('2023-02-07 00:00:00.000000', 'YYYY-MM-DD HH24:MI:SS.US')
GROUP BY v1
ORDER BY count DESC
LIMIT 10000
```

### To Reproduce

```
dev=> create table test (v1 timestamp);

dev=> select * from test where v1 > TO_TIMESTAMP('2023-02-06 00:00:00', 'YYYY-MM-DD HH24:MI:SS');
ERROR:  QueryError: internal error: Failed to build executor: Expr error: Unsupported function: Timestamp cmp Varchar
```

**Comment by xiangjinwu** [comment]
We hard-coded `to_timestamp` to return `varchar` by mistake.

https://github.com/risingwavelabs/risingwave/blob/d8198fa138003e1f1431053f4f5f09e4a5fa8fd8/src/expr/src/expr/expr_to_timestamp_const_tmpl.rs#L37-L40

Should be `timestamp` to match current implementation.
Should be `timestamptz` and update implementation to align with PostgreSQL #7780.

**Comment by xiangjinwu** [comment]
It is being fixed to `Timestamp` first: #9924
